audio-worklet/src/noisegate-script-processor.js

```diff
@@ -108,11 +108,11 @@ class NoiseGate {
 
     let envelope = this.detectLevel_(this.channel_);
     envelope = this.convertEnvelopeToSineNormalizedDecibelScale_(envelope);
-
+    let weights = this.computeWeights_(envelope);
```

Comment:
Oh, and definitely agree with `computeWeighs_` not modifying the input.  Which you could have avoided by doing by extracting each element and scaling it and using it in the comparisons, without modifying `envelope`.